dataclients/kubernetes/kube.go

```diff
@@ -467,11 +467,19 @@ func (c *Client) ingressToRoutes(items []*ingressItem) ([]*eskip.Route, error) {
 				if prule.Backend.Traffic > 0 {
 					r, err := c.convertPathRule(i.Metadata.Namespace, i.Metadata.Name, rule.Host, prule, servicesURLs)
 					if err != nil {
-						// tolerate single rule errors
+						// fail when there's a temporary network error so that we keep the last known rules
+						// active until the next sync loop.
+						if err, ok := err.(net.Error); ok {
```

Comment:
I think this is too low-level. Imagine the apiserver respond with 500 then this would be treated as a permanent error.